chainerx_cc/chainerx/cuda/cuda_device/arithmetic.cu

```diff
@@ -281,6 +281,105 @@ public:
 
 CHAINERX_CUDA_REGISTER_KERNEL(PowerSAKernel, CudaPowerSAKernel);
 
+// CUDA does not have std::mod, which is used for the native backend.
+template <typename T>
+__device__ T ModSignedIntegerImpl(T x, T y) {
```

Comment:
The current implementation seems a little complicated. How about the following one?
```cpp
if (x == 0 || y == 0) {
    return 0;
}
T ret = x % y;
if (ret > 0 && y < 0) {
    return ret - y;
}
if (ret < 0 && y > 0) {
    return ret + y;
}
return ret;
```